qiskit/algorithms/gradients/reverse_gradient/reverse_gradient.py

```diff
@@ -0,0 +1,203 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Estimator gradients with the classically efficient reverse mode."""
+
+from __future__ import annotations
+from collections.abc import Sequence
+import logging
+
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit, Parameter
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info import Statevector
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import Estimator
+
+from .bind import bind
+from .derive_circuit import derive_circuit
+from .split_circuits import split
+
+from ..base_estimator_gradient import BaseEstimatorGradient
+from ..estimator_gradient_result import EstimatorGradientResult
+from ..lin_comb_estimator_gradient import DerivativeType
+
+logger = logging.getLogger(__name__)
+
+
+class ReverseEstimatorGradient(BaseEstimatorGradient):
+    """Estimator gradients with the classically efficient reverse mode.
+
+    .. note::
+
+        This gradient implementation is based on statevector manipulations and scales
+        exponentially in numbers of qubits. However, for small system sizes it can be very fast
+        compared to circuit-based gradients.
+
+    This class implements the calculation of the expectation gradient as described in
+    [1]. By keeping track of two statevectors and iteratively sweeping through each parameterized
+    gate, this method scales only linearly in the number of parameters.
+
+    **References:**
+
+        [1]: Jones, T. and Gacon, J. "Efficient calculation of gradients in classical simulations
+             of variational quantum algorithms" (2020).
+             `arXiv:2009.02823 <https://arxiv.org/abs/2009.02823>`_.
+
+    """
+
+    SUPPORTED_GATES = ["rx", "ry", "rz", "cp", "crx", "cry", "crz"]
+
+    def __init__(self, derivative_type: DerivativeType = DerivativeType.REAL):
+        """
+        Args:
+            derivative_type: Selects whether the real, imaginary or real plus imaginary part
+                of the gradient is returned.
+        """
+        alibi_estimator = Estimator()  # this is never used
+        super().__init__(alibi_estimator)
```

Comment:
`alibi`? hahaha maybe `dummy` would be clearer

```suggestion
        dummy_estimator = Estimator()  # this is required by the base class, but not used
        super().__init__(dummy_estimator)
```